boot-qemu.sh

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env bash
+
+# Root of the repo
+BASE=$(cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")" && pwd)
+
+
+# Prints an error message in bold red then exits
+function die() {
+    printf "\n\033[01;31m%s\033[0m\n" "${1}"
+    exit 1
+}
+
+
+# Check that a binary is found
+function checkbin() {
+    command -v "${1}" &>/dev/null || die "${1} could not be found, please install it!"
+}
+
+
+# Parse inputs to the script
+function parse_parameters() {
+    while (( ${#} )); do
+        case ${1} in
+            -a|--arch|--architecture)
+                shift
+                case ${1} in
+                    arm32_v5|arm32_v6|arm32_v7|arm64|mips|mipsel|ppc32|ppc64|ppc64le|x86_64) ARCH=${1} ;;
+                    *) die "Invalid --arch value '${1}'" ;;
+                esac ;;
+
+            -d|--debug)
+                set -x ;;
+
+            -h|--help)
+                echo
+                cat "${BASE}"/boot-qemu-help.txt
+                echo
+                exit 0 ;;
+
+            -i|--interactive|--shell)
+                INTERACTIVE=true ;;
+
+            -k|--kbuild-folder)
+                shift && KBUILD_DIR=${1} ;;
+
+            -t|--timeout)
+                shift && TIMEOUT=${1} ;;
+
+            *)
+                die "Invalid parameter '${1}'" ;;
+        esac
+        shift
+    done
+}
+
+
+# Sanity check parameters and required tools
+function sanity_check() {
+    # Kernel build folder and architecture are required paramters
+    # We are not interactive by default and the default timeout is 3m
+    : "${ARCH:?}" "${KBUILD_DIR:?}" "${INTERACTIVE:=false}" "${TIMEOUT:=3m}"
```

Comment:
what does the `:` do?

Reply:
It is basically a shell builtin that acts like `true`. The point here is so that I can avoid a bunch of:

```
[[ -z ${ARCH} ]] && die "ARCH is undefined"
```

and such.

Reply:
But you only need it for `ARCH` and `KBUILD_DIR` because of the `:?`?

When we start using sigils, I get nervous that the code starts to look like Perl, and there's good jokes about Perl being unreadable for a reason.

Reply:
Yes, I prefer to be explicit about the variables that I expect to be defined rather than using `set -u`.

Otherwise, I can do the more verbose thing I mentioned if it is more readable.

Reply:
+1 @nickdesaulniers 

Specially if the symbols are not that common in other languages.

Reply:
Alright, I pushed 936f954458982b8273b470bbfd4b05547051bd06, which should be less cryptic.

Reply:
That looks way better for both user and dev, thanks!